### PR TITLE
Feature/acl 18 select list

### PR DESCRIPTION
**Pull request by aaa-magenic-eleazarj**
### Links
https://autoclub.atlassian.net/browse/ACL-18

### Description
A SelectList is a base component for something like a SelectDropdown or RadioGroup. This pr is for primary type only (string/number type of display) 

### How to test
yarn storybook 

### Additional Notes
Additional Unit tests to follow in a separate pr

### Explanation of Jest/Unit Testing
Tests included are only for SelectListItemText component. Will include additional test for SelectList component.

### CODE REVIEW

File: src/lib/components/SelectListItemText/SelectListItemText.test.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  describe('html rendering', () => {
+    it('rendered text should match passed options text', () => {
+      props = getFakeProps({
+        item: { id: 1, value: 1, display: 'I am Iron Man' }
+      });
+
+      listItemTextWrapper = createSelectListItemTextWithTheme(props);
+
+      const listItem = listItemTextWrapper
+        .find('.MuiListItemText-primary')
```
Comment (ElGrecode) [inline src/lib/components/SelectListItemText/SelectListItemText.test.js]: Is this possible to find by React component
```
import MUIListItem
.find(MUIListItem)
```

File: stories/SelectList.stories.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    value: 3,
+    display: 'Third Item '
+  },
+  { id: 4, value: 4, display: '4th Item' },
+  { id: 5, value: 5, display: '5th Item' },
+  {
+    id: 6,
+    value: 6,
+    display: '6th item'
+  },
```
Comment (ElGrecode) [inline stories/SelectList.stories.js]: Is there a reason some are spread to 3 lines vs 1?

File: src/lib/components/SelectList/SelectList.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                    dense
+                    className={cx('List', classes.root, {
+                      [classes.fullOverlay]: items.length > 6
+                    })}
+                  >
+                    {items.map(item => (
+                      <SelectListItemText
+                        key={item.id}
+                        item={item}
+                        selected
```
Comment (ElGrecode) [inline src/lib/components/SelectList/SelectList.js]: Does this mean it always passes a prop of selected?

File: src/lib/components/SelectListItemText/SelectListItemText.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  primary: {
+    fontSize: '18px',
+    letterSpacing: 'normal',
+    fontStretch: 'normal',
+    lineHeight: 1.5,
+    [theme.breakpoints.between('xs', 'sm')]: {
+      fontSize: '16px'
+    }
+  },
+  selected: {}
```
Comment (ElGrecode) [inline src/lib/components/SelectListItemText/SelectListItemText.js]: Do we need this if it's always empty?

File: src/lib/components/SelectList/SelectList.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      disabled?: PropTypes.bool
+    }
+  ],
+  type: PropTypes.string,
+  onSelect: PropTypes.func
+};
+
+const styleClasses = theme => ({
+  root: {
+    width: '341px',
```
[the remark this answers is not in the document]
Comment (aaa-magenic-eleazarj) [reply src/lib/components/SelectList/SelectList.js]: Fixed. Also applied this to most of the attributes that has 'px'

File: src/lib/components/SelectList/SelectList.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    background: theme.palette.common.white,
+    border: `2px solid ${theme.palette.primary.main}`,
+    borderRadius: '4px',
+    padding: '0px',
+    boxShadow: '0 2px 8px 0 rgba(0, 0, 0, 0.35)',
```
[the remark this answers is not in the document]
Comment (aaa-magenic-eleazarj) [reply src/lib/components/SelectList/SelectList.js]: Fixed. reused GRAY

File: src/lib/components/SelectListItemText/SelectListItemText.test.js (hunk 1 above)
Comment (aaa-magenic-eleazarj) [reply src/lib/components/SelectListItemText/SelectListItemText.test.js]: Used span instead

File: stories/SelectList.stories.js (hunk 2 above)
Comment (aaa-magenic-eleazarj) [reply stories/SelectList.stories.js]: No. I think they are autoformatted upon saved. Fixed.

File: src/lib/components/SelectListItemText/SelectListItemText.js (hunk 4 above)
Comment (aaa-magenic-eleazarj) [reply src/lib/components/SelectListItemText/SelectListItemText.js]: Removed

File: src/lib/components/SelectList/SelectList.js (hunk 3 above)
Comment (aaa-magenic-eleazarj) [reply src/lib/components/SelectList/SelectList.js]: Removed

File: src/lib/components/SelectListItemText/SelectListItemText.test.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+  describe('html rendering', () => {
+    it('rendered text should match passed options text', () => {
+      props = getFakeProps({
+        item: { id: 1, value: 1, display: 'I am Iron Man' },
+      });
+
+      listItemTextWrapper = createSelectListItemTextWithTheme(props);
+
+      const listItem = listItemTextWrapper.find('span').at(0);
```
Comment (ElGrecode) [inline src/lib/components/SelectListItemText/SelectListItemText.test.js]: ahh, I think if you're just trying to get the text here-- Enzyme has a helper here `.text()`

File: src/lib/components/SelectListItemText/SelectListItemText.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+import ListItemText from '@material-ui/core/ListItemText';
+
+type propTypes = {
+  // Decorator Props
+  classes: PropTypes.object,
+  // Passed Props
+  item: {
+    id: PropTypes.string | PropTypes.number,
+    value: PropTypes.string | PropTypes.number,
+    display: PropTypes.string | PropTypes.number
```
Comment (ElGrecode) [inline src/lib/components/SelectListItemText/SelectListItemText.js]: Can we add a prop type of selected here as well. 
`selected?: PropTypes.bool`

File: src/lib/components/SelectListItemText/SelectListItemText.test.js (hunk 7 above)
Comment (ElGrecode) [reply src/lib/components/SelectListItemText/SelectListItemText.test.js]: Ignore. I see there is another PR for unit tests.

File: src/lib/components/SelectListItemText/SelectListItemText.js (hunk 8 above)
Comment (aaa-magenic-eleazarj) [reply src/lib/components/SelectListItemText/SelectListItemText.js]: Added

File: src/lib/components/SelectList/SelectList.test.js [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+      selectListWrapper = createSelectListWithTheme(props);
+
+      const listItems = selectListWrapper.find('li');
+
+      listItems.forEach((listItem, i) => {
+        expect(listItem.text()).to.equal(props.items[i].display);
+      });
+    });
+
+    it('should not render items is no primary type is passed', () => {
```
Comment (aaa-magenic-tedb) [inline src/lib/components/SelectList/SelectList.test.js]: typo error in the description. - render items if...

**Pull request merged**